src/BaseModel.php

```diff
@@ -256,7 +256,15 @@ public static function prepareAssocFromRequest(array $row): array
     public static function select($select = ['*']): Builder
     {
         $instance = new static();
-        return (new Builder($instance->getThisClient()))->select($select)->from($instance->getTable());
+        return $instance->newQuery()->select($select)->from($instance->getTable());
+    }
+
+    /**
+     * @return Builder
+     */
+    protected function newQuery(): Builder
+    {
+        return new Builder($this->getThisClient());
```

Comment:
Let's encapsulate `->from($instance->getTable())` here.

Reply:
The purpose of encapsulating a builder is to be able to override it with your own builder. This will make it possible to define scopes and extend the behavior of the default builder